### ISSUE TITLE
[Feature request] Package/Port for FreeBSD

### ISSUE THREAD

**Issue by Peter2121** [ONLYOFFICE/DesktopEditors#33]
Please, prepare packages/ports of community edition (server) and desktop edition for FreeBSD.

**Comment by maxkadushkin** [comment]
do you mean [CommunityServer](https://github.com/ONLYOFFICE/CommunityServer) and DesktopEditors? there is portable version of editors. is it suitable for you?

**Comment by Peter2121** [comment]
The most interesting thing is DesktopEditors.
The portable version is built for Linux x64, not for FreeBSD. Maybe, using Linux x64 compatibility layer it could be run on FreeBSD 11. Certainly, it does not work on FreeBSD 10.3 (my workstation), probably because x64 ELF files cannot be run on this version of FreeBSD. Anyway, using Linux binaries on FreeBSD stays tricky, it is not intended for end user. I don't think that recompiling OnlyOffice DesktopEditors for FreeBSD could be really difficult, as it seems that all the components are available on FreeBSD.
So, please, recompile it for FreeBSD.

**Comment by beatgammit** [comment]
Or do the work necessary to get it into the ports system.

**Comment by GaroRobe** [comment]
I'll give my own input (sorry for dramatic presentation, I've had some unhappy hours trying to make it work):

I've encountered a pretty viable scenario when I needed CommunityServer on FreeBSD. Here's why:

1. I've hacked up home NAS for family and personal/freelance use.
2. The most popular NAS for the task is FreeNAS - which is FreeBSD-based.
3. Having a personal cloud for collaboration projects is very sueful - thus, NextCloud on FreeNAS...
4. There is ONLYOFFICE integration for NextCloud which allows me and my family/collegues use Google Sheets/Text/Whatever-like functionality without google. On my infrastructure. Which is good.
5. <OOPS!> ONLYOFFICE wants CommunityServer. :(
6. I did managed to launch rancher+docker+CommunityServer and configure it. AND IT WORKED!!! :)
7. But if only it would be possible to run Server natively - then I could make a complete, good FreeNAS OO CommunityServer plugin - and all the pain would go away in 1 click (yep, rancher didn't survive system upgrade, while plugins did and I'm not happy with it)

Some could say "You've picked a bad OS - not OO problem". 
My position is - "Would you like to get FreeNAS users - make it work in jail".

**Comment by ShockwaveNN** [comment]
@GaroRobe I think your comment is misplaced, it should be in https://github.com/ONLYOFFICE/DocumentServer/issues/79 or new issue in https://github.com/ONLYOFFICE/CommunityServer/issues if you really mean CommunityServer, not DocumentServer

**Comment by pixel8383** [comment]
Any news on this?

**Comment by ShockwaveNN** [comment]
@pixel8383 Sorry, no news

**Comment by GimmeLewd** [comment]
We need ONLYOFFICE for *BSD, please.

**Comment by chlarsen** [comment]
+1

**Comment by Teetoow** [comment]
I spent many hours trying to only compile the DocumentServer part of onlyoffice.
It was nearly a success before I discovered that onlyoffice needs some node.js packages which are obsolete, unmaintained and thus unavailable for FreeBSD.
Before that I had to update tens and tens of source code files, Makefiles etc. in order to get it compiling.
This work is really boring and can't be finished until OnlyOffice uses still maintained external packages.
Moreover I can't believe that someone will accept to do this job each time a new release of OnlyOffice is out.
The only solution would be that OnlyOffice accepts these patches in order to natively support FreeBSD.

Last but not least I tried many ways in order to use the linux binaries into FreeBSD through the compat layer. It does not works.

At the end there is two solution: 
 - the ugly one: use virtualbox with a linux distribution in order to get it working
 - the best one: convince  OnlyOffice to natively support FreeBSD (but I don't think they'll accept even if we send to them patches for that purpose..)

**Comment by Peter2121** [comment]
Happy to see that someone is trying to build it.
I'm surprised to know that OnlyOffice uses unmaintained node.js packages. I think that you can open an issue about the subject, as unmaintained=insecure, and it is not really good for a web application.
IMHO, we cannot know if OnlyOffice accepts FreeBSD compatibility patches if nobody sends them ;)

**Comment by Teetoow** [comment]
Yeah you're right @Peter2121. I stopped trying to work on it because of a lack of time and these node.js problems. It was the last summer.. Now I had to redo the job from the beginning..
I also sent a feature request to DocumentServer authors. I'll see their answer.
I won't be able to try it again before this summer.. It's definitely a big job to do.
Or I hope that someone else will do it :)

**Comment by Teetoow** [comment]
Follow the FreeBSD patches I wrote here: https://github.com/ONLYOFFICE/DocumentServer/issues/79

**Comment by Tectu** [comment]
@Teetoow Thank you for your efforts! May I inquire information about the current state of getting the DocumentServer and the desktop software running on FreeBSD?

**Comment by Teetoow** [comment]
DocumentServer works like a Charm under FreeBSD with my patches. I use it everyday.
About desktop software I don't know (and I don't think it will compile) since I don't use it (and as far as I can see, I'm the only one who try to get patches for FreeBSD..)

**Comment by srdjanpavlovic** [comment]
My only reason staying on Linux now is OnllyOffice DesktopEditors have no port for FreeBSD, and I find it most compatible with MS formats... please, someone make FreeBSD port...

**Comment by Tectu** [comment]
> please, someone make FreeBSD port...

I can certainly help with the port itself. However, I assume that getting upstream to build on FreeBSD is going to be the time consuming effort. Any takers?

**Comment by andrewhotlab** [comment]
I'm very interested in getting [ONLYOFFICE/DesktopEditors](https://github.com/ONLYOFFICE/DesktopEditors) ported to FreeBSD too.
Reading instructions [here](https://github.com/ONLYOFFICE/build_tools#desktop-editors), I tried to build on a FreeBSD machine, but [ONLYOFFICE/build_tools](https://github.com/ONLYOFFICE/build_tools) at this moment are only available for these platforms: https://github.com/ONLYOFFICE/build_tools/tree/master/tools.

Thus I guess we need to port Linux build tools to FreeBSD. I am not able to do this, neither I know someone who can, but I can sponsor the work of someone who is.
If you are or know someone able to do that, please contact me to talk about it, thanks!

**Comment by Zagzigger** [comment]
I would be very interested in using onlyoffice on FreeBSD. I've used it on other platforms, and it is a very well thought out product. Furthermore, I'm sure many others would welcome it as an application.

**Comment by Tectu** [comment]
I'd welcome OnlyOffice on FreeBSD too.
In the meantime, I have become a FreeBSD ports committer. However, I'll be extremely busy for the next few months.
If somebody wants to get this effort started, I'll gladly help where I can.
Not sure if @andrewhotlab 's offer is still on the table.

**Comment by andrewhotlab** [comment]
Yes I'm still interested: thank you @Tectu, and my sincere congratulations for your role as FreeBSD contributor!  ! I'll contact you directly ASAP from my personal email address in order to check how I can support your work. Thanks!

**Comment by andrewhotlab** [comment]
Hi @Tectu, I wrote you a month ago... if you are still interested, you can contact me directly whenever you'll be able. Thanks.
